sky/data/storage.py

```diff
@@ -60,6 +61,19 @@
     'It may have been deleted externally.')
 
 
+def _is_storage_cloud_enabled(cloud_name: str,
+                              try_fix_with_sky_check: bool = True) -> bool:
+    enabled_storage_clouds = global_user_state.get_enabled_storage_clouds()
+    if cloud_name in enabled_storage_clouds:
+        return True
+    if try_fix_with_sky_check:
```

Comment:
I'm not fully sure if it is a good idea to run sky check if the storage cloud is not already in `get_enabled_storage_clouds()`. Running this adds a latency of ~7s:
```
>>> s=time.time();sky.check.check(quiet=True);print(time.time()-s)
7.19357705116272
```
Though this latency hit would not be frequent, I wonder if there's some alternate solution. Can we run `sky check` on the spot controller at the end of its setup? 

I guess this also relates to semantics of `sky check`. Do we expect it to be run once by the user, or is it acceptable to be run frequently by our code? For example, I often need to disable a specific cloud. To do so, I temporarily move it's credentials file from it's path (e.g., `mv ~/.aws/credentials /tmp/credentials`) and then run `sky check`. That disables the cloud, and now I can move my credentials file back so I can continue using aws cli tools if I need to. If SkyPilot calls sky.check() quietly in the code, aws would get enabled again (which is not what I would want to happen).

Reply:
That is a good point!
Running the `sky check` at the end of its setup will cause the overhead occurring every time the spot job is submitted, which could make the latency more significant.

Our optimizer has the same logic for re-checking the enabled cloud when the cloud is not enabled, so doing it here may not make a significant difference from before. https://github.com/skypilot-org/skypilot/blob/072874bcc5589824ddfe0ab7bc4f9f936702d2b8/sky/optimizer.py#L943-L949

An optimization we could do here might be just to check for the cloud that is specified. Wdyt?

For the use case of moving the credential and back, I find the original method might be quite hacky already, since the user may not have the idea of the credential files. For that, we may want to offer an explicit way to allow the user to disable some clouds.

Reply:
This code looks good to me, because:
- If a user has not called `sky.check()` already
  - Before this PR: a `sky.check()` is automatically called by `sky launch -> optimizer` code path anyway
  - After this PR: a `sky.check()` is possibly called at YAML parsing time (if it specifies a store; storage.py is eager); if so, afterwards the `sky launch -> optimizer` code path doesn't need to re-incur this overhead

So it seems like the overhead is incurred once before & after this PR, if users have not called `sky.check()`.

Please correct if this is wrong @Michaelvll.